### PR TITLE
introduce github workflow on pull request

### PR DESCRIPTION
**Pull request by annarieger**
Introduce a basic workflow to check the code changes on a pull request.

### CODE REVIEW

File: .github/workflows/on-pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Test app on a new pull request
+
+on:
+  push:
+    branches: [ master, ol5 ]
+  pull_request:
+    branches: [ master, ol5 ]
```
Comment (dnlkoch) [inline .github/workflows/on-pull-request.yml:7]: Not related to this PR in particular, but can we please rename the branch to `main`?

File: .github/workflows/on-pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x]
```
Comment (dnlkoch) [inline .github/workflows/on-pull-request.yml:15]: What about `v16`?

File: .github/workflows/on-pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    - name: Cache Node.js modules
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.OS }}-node-
+          ${{ runner.OS }}-
+    - name: Install dependencies
```
Comment (dnlkoch) [inline .github/workflows/on-pull-request.yml:35]:
```suggestion

    - name: Install dependencies
```

File: .github/workflows/on-pull-request.yml (hunk 2 above)
Comment (dnlkoch) [reply .github/workflows/on-pull-request.yml:15]: Ok, too bad.

**Comment by annarieger** [comment]
Superseded by #406

**Pull request closed without merging**